image-classification-with-distributed-training/horovod-project.ipynb

```diff
@@ -5,92 +5,203 @@
+    "To increase the likelihood of estimating the actual gradient we could use bigger batches, by sending small batches to different GPUs running the same Neural Network, calculating the batch gradient and then running a **Synchronization Step** to calculate the average gradient over the batches and update the Neural Networks running on the different GPUs.  \n",
+    "\n",
+    "\n",
+    "> It is important to understand that the act of distribution adds extra **Synchronization Costs** which may vary according to your cluster's configuration.  \n",
+    "> <br>\n",
+    "> As the gradients and NN needs to be propagated to each GPU in the cluster every epoch (or a number of steps), Networking can become a bottleneck and sometimes different configurations need to be used for optimal performance.  \n",
+    "> <br>\n",
+    "> **Scaling Efficiency** is the metric used to show by how much each additional GPU should benefit the training process with Horovod showing up to 90% (When running with a well written code and good parameters).\n",
+    "\n",
+    "![Horovod scaling](https://user-images.githubusercontent.com/16640218/38965607-bf5c46ca-4332-11e8-895a-b9c137e86013.png)\n",
+    "\n",
+    "### How can we distribute our training\n",
+    "There are two different cluster configurations (which can be combined) we need to take into account.  \n",
+    "- **Multi Node** &mdash; GPUs are distributed over multiple nodes in the cluster.  \n",
+    "- **Multi GPU** &mdash; GPUs are within a single Node.  \n",
+    "\n",
+    "In this demo we show a **Multi Node Multi GPU** &mdash; **Data Parallel** enabled training using Horovod.  \n",
+    "However, you should always try and use the best distribution strategy for your use case (due to the added costs of the distribution itself, ability to run in an optimized way on specific hardware or other considerations that may arise).\n",
+    "\n",
+    "### How Horovod works?\n",
+    "Horovod's primary motivation is to make it easy to take a single-GPU training script and successfully scale it to train across many GPUs in parallel. This has two aspects:\n",
+    "\n",
+    "- How much modification does one have to make to a program to make it distributed, and how easy is it to run it?\n",
+    "- How much faster would it run in distributed mode?\n",
+    "\n",
+    "Horovod Supports TensorFlow, Keras, PyTorch, and Apache MXNet.\n",
+    "\n",
+    "in MLRun we use Horovod with MPI in order to create cluster resources and allow for optimized networking.  \n",
+    "**Note:** Horovd and MPI may use [NCCL](https://developer.nvidia.com/nccl) when applicable which may require some specific configuration arguments to run optimally.\n",
+    "\n",
+    "Horovod uses this MPI and NCCL concepts for distributed computation and messaging to quickly and easily synchronize between the different nodes or GPUs.\n",
+    "\n",
+    "![Ring Allreduce Strategy](https://miro.medium.com/max/700/1*XdMlfmOgPCUG9ZOYLTeP9w.jpeg)\n",
```

Comment:
I think we should split some of the text to separate markdown cells. Users in Jupyter will be able to "run" each cell. Otherwise they get a large one that they need to scroll and cannot use the keyboard

Reply:
Split the markdown cells by headlines.